### DISCUSSION TITLE
Make javacc-8 to be the main repo and freeze javacc-7

### DISCUSSION THREAD

**Discussion by new-javacc** [javacc/javacc#240] [General]
Freeze old javacc as javacc-7 and move all new development to the main javacc repo. javacc-8 as a repo is a bit odd because it's a particular version and when we want to do a new major version it could be weird to create another repo. Also making it the main repo will make sure all new development goes here.

CC: @mbrigl @zosrothko @vlsi

**Comment by vlsi** [comment]
I don't understand what do you mean by "Freeze old javacc as javacc-7". Would you please clarify? Are you going to rename `javacc/javacc` repository? Are you going to keep its history? Are you going to rename `javacc-8`?

What is the purpose of freezing anything?

**Comment by new-javacc** [comment]
Fine - we don't need to freeze anything physically but by main goal is to move all active development to 8 which is a major refactor with the core separated from code generators.

**Reply by vlsi** [reply]
What do you mean by 8?
Frankly speaking, I find that https://github.com/javacc/javacc-8-core/issues/1 is really a blocker to continue using javacc-8-core history. I believe it is wrong to continue working on those sources before the code has proper git history.

**Reply by zosrothko** [reply]
I do not see why the missing git history is a blocker point?? The new development could start from a given level of source code without history.

**Reply by mbrigl** [reply]
It might not be a blocker, but it is bad choice. 
Frankly speaking, already the outsourcing of javacc 8 destroyed a lot of information. Redesign of JavaCC 8 was started as a branch of JavaCC. At some time the branch has been "copied" to a new repository and the old branch has been deleted from repository.  All the transition history has been lost forever.
I agree that the JavaCC should be a new version (branch) of the original repository. JavaCC 8 is a redesign, not a rewrite of JavaCC.

**Reply by zosrothko** [reply]
@mbrigl I do not understand your point. Unles a mistake from my side, the history since 
```
Author: tigrisc <tigrisc@68c0822f-77d1-6f10-b115-8a14d106ca4c>  2003-06-06 00:13:50
Committer: tigrisc <tigrisc@68c0822f-77d1-6f10-b115-8a14d106ca4c>  2003-06-06 00:13:50
Child:  0656aab94ec87efb6875290b4a20939fe6bd0ec6 (Initial import)
Branches: release, remotes/origin/Token-with-virtual-methods, remotes/origin/check_epsilon_move_statename, remotes/origin/master, remotes/origin/release
Follows: 
Precedes: 

    Commit by Helm::Ccvs

```
is keep both in [javacc](git@github.com:javacc/javacc.git) and [javacc-8](git@github.com:javacc/javacc-8.git).

Could you please confirm or infirm this point?

**Reply by mbrigl** [reply]
Hello @zosrothko, sorry you are right. I checked again, the history has been imported in the main repository JavaCC8. But there the history is quite useless. All the source code (core and generators) are in the sub-modules and the file history is not available anymore.

**Comment by martinswanson** [comment]
Francis - if I understand correctly the recommendation from the 3 of us is to continue to use the original javacc repo to preserve the history there. The javacc-8 work could be copied into that repo as a branch, refactored with submodules, then released as the “current” version. That would in effect stop development on version 7 (altho bug fixes would still be allowed, as normal).

Is there any reason not to do that ?

**Comment by new-javacc** [comment]
The code generators are completely new so no history will be relevant. IMO  we are wasting too much time on it.
Hello @zosrothko, sorry you are right. I checked again, the history has been imported in the main repository JavaCC8. But there the history is quite useless. All the source code (core and generators) is in the sub-modules and the file history is not available anymore.

**Reply by zosrothko** [reply]
The git history from [https://github.com/javacc/javacc](https://github.com/javacc/javacc) i.e. javacc-7 has been imported into the [javacc-8-core](https://github.com/javacc/javacc-8-core) repo under the name javacc-7/. Everybody can sse the import on [contributors](https://github.com/javacc/javacc-8-core/graphs/contributors). Hope this could unblock volunters and contributors as well a bringing some peace here.

**Reply by new-javacc** [reply]
> The git history from https://github.com/javacc/javacc i.e. javacc-7 has been imported into the [javacc-8-core](https://github.com/javacc/javacc-8-core) repo under the name javacc-7/. Everybody can sse the import on [contributors](https://github.com/javacc/javacc-8-core/graphs/contributors). Hope this could unblock volunters and contributors as well a bringing some peace here.

Thank you very much for all your efforts in this, @zosrothko! And hopefully with this we will move to the new world where codegens improve tremendously fast without creating troubles for others!

**Reply by martinswanson** [reply]
You're talking about just the commit history? 
What about the history of forks, stars, used by, and committers ? AFAICS none of this information appears in the javacc-8 repo.

**Comment by zosrothko** [comment]
> What about the history of forks, stars, used by, and committers ? AFAICS none of this information appears in the javacc-8 repo
Up to now, July 11 2022, there are 0 fork on any javacc-8* repository except javacc-8-java which has one fork.

**Comment by martinswanson** [comment]
Precisely. You lost all the history from the javacc repo.

**Comment by martinswanson** [comment]
I think you missed the point.

You’re forcing ~600 people who have already forked the javacc repo and set up local git versions of their forks to now delete all that, and then fork ~3-4 brand new repos. If you just pushed back to a branch in the original repo there would be no impact on users, and you’d preserve all the history of forks, stars, used by, and contributors, rather than wiping it clean.

**Reply by new-javacc** [reply]
Actually I wanted to wipe it clean with the understanding that if you want the new stuff you need to create a new fork. This is more than a redesign. This is actually almost a rewrite.

**Reply by new-javacc** [reply]
I don't mind keeping that old javacc around if users want to continue to work on it.  But the way forward is the new way if a bit painful one might actually enjoy working on the new code generators which is where 100% of contributions were over the last 5+ years. I can count on my fingers the PRs that are core rfe or bug fixes.

**Comment by martinswanson** [comment]
-> Up to now, July 11 2022, there are 0 fork on any javacc-8* repository 

I guess no-one wants the new stuff.

**Comment by new-javacc** [comment]
On Jul 11, 2022 3:59 PM, Martin Swanson ***@***.***> wrote:

-> Up to now, July 11 2022, there are 0 fork on any javacc-8* repository 



I guess no-one wants the new stuff.Fine by me! I want it for myself lol. I fully intend to build js and py code generators using this. That's one area where we can get parity with ANTLR. And it's very important.
